README.md

```diff
@@ -1,2 +1,43 @@
 # cross-capi-metal
 
+This repository contains a guide and related manifests for using
+[Crossplane](https://github.com/crossplaneio/crossplane) and [Cluster
+API](https://github.com/kubernetes-sigs/cluster-api) with infrastructure on
+[Equinix Metal](https://metal.equinix.com/).
+
+## Repository Contents
+
+- Simple Golang echo app that can be run in client or server mode.
+- Helm chart to install app into a Kubernetes cluster.
+- Crossplane package to deploy echo server on Equinix Metal.
+
+## Guide
+
+1. Create a control plane cluster. This can be any Kubernetes cluster. We
+   recommend starting out with [KIND](https://kind.sigs.k8s.io/).
+2. Install CAPI's
+   [clusterctl](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl).
+3. Go through the [CAPI
+   quickstart](https://cluster-api.sigs.k8s.io/user/quick-start.html) to
+   provision a cluster on Equinix Metal (still called `Packet` in CAPI docs).
```

Comment:
@detiber ideally we skip the `clusterctl` quickstart and instead put the types that get created into a composition so we can do everything in what is currently step 8. Obviously there would be some setup still required to have CAPI and the equinix metal provider present, but once that is true do you know what CRs are created (i.e. what we could move to the composition)?

Reply:
There should be a way that we can dump out all the yaml that is applied by clusterctl, will work up the details.

Reply:
Dumping provider manifests:
```sh
clusterctl config provider --core=cluster-api -o yaml > core.yaml
clusterctl config provider --bootstrap=kubeadmi -o yaml > kubeadm-bootstrap.yaml
clusterctl config provider --control-plane=kubeadmi -o yaml > kubeadm-control-plane.yaml
clusterctl config provider --infrastructure=packet -o yaml > packet-infrastructure.yaml
```

The output above will need to be run through envsubst before being applied, the main env var that we'd need to worry about is PACKET_API_KEY, we should be able to rely on the defaults for the rest.

Prior to applying the above yaml, cert-manager will need to be deployed on the management cluster.

`clusterctl config cluster` can be used similar to the quickstart to pre-generate the yaml needed for creating a cluster, unlike `clusterctl init` it doesn't operate directly against a k8s cluster it just outputs the processed template to stdout.

Once the cluster yaml is applied, then you should be able to look for the presence of the kubeconfig secret, which will live in the same namespace as the cluster created and be named `<cluster name>-kubeconfig`.

Once the kubeconfig is retrieved, it can be used to poll the remote cluster for availability, once it can be reached, then the CNI can be deployed (cluster configuration defaults are aligned with calico).

Once CNI is deployed, any additional control plane instances and worker node instances will be able to be created.

Reply:
I think https://cluster-api.sigs.k8s.io/user/quick-start.html#generating-the-cluster-configuration is the main step we want to do in composition, the rest can just be setup beforehand